### ISSUE TITLE
document that StartLimitBurst=/StartLimitIntervalSec='s effect is mooted if the unit is unloaded, due to the GC logic

### ISSUE THREAD

**Issue by fbuihuu** [systemd/systemd#7139]
According to my understanding of the man pages, the following should fail:

```
# systemctl cat test.service
# /etc/systemd/system/test.service
# cat /etc/systemd/system/test.service
[Unit]
Description=test

[Service]
ExecStart=/usr/bin/true

# systemctl show test.service | grep StartLimit
StartLimitIntervalSec=10000000
StartLimitBurst=5
StartLimitAction=none

# for i in $(seq 7); do systemctl start test.service; sleep 0.5; done
```

since test.service is started more than 5 times within 10 sec.

However the unit can be started 7 times without any error.

This was tested with v234 (v228 shows the same behavior though).

**Comment by poettering** [comment]
Well, if the service is unloaded in between then we forget everything about the start counter. And in this case it will be unloaded, since nothing references the unit, and because it doesn't fail.

i figure we should document this though...

**Comment by fbuihuu** [comment]
I can understand why the previous test case doesn't fail but that makes the whole StartLimit feature unpredictable and mostly useless for the start operation. 

The same behavior is seen for the following sequence BTW:

`# for i in $(seq 7); do systemctl start sleep.service && systemctl sleep.service; done`

In contrary, the limit can be reached whereas it isn't expected: someone reported the [case](https://bugzilla.suse.com/show_bug.cgi?id=1060734) where the systemd user instance was started and stopped sequentially (by using sudo) several times and the start limit was reached.

So for example, doing:

```
# sudo rpm -V package1
# sudo rpm -V package2
# ...
# sudo rpm -V package9
```

can trigger the limit mostly because the stop action of the user instance is asynchronous...

Maybe we should limit this feature to the restart one only instead ?

**Comment by fbuihuu** [comment]
@poettering ? this looks more like a bug to me.

**Comment by poettering** [comment]
Fix waiting in #7337
